UTY-2661: Improve snapshot api (#1479)
spatialos/gdk-for-unity: CHANGELOG.md

- The minimum supported version of Unity is now 2020.1. [#1459](https://github.com/spatialos/gdk-for-unity/pull/1459)
- Upgraded Unity Entities to 0.14.0-preview.18. [#1463](https://github.com/spatialos/gdk-for-unity/pull/1463)
  - Projects now require the `Unity Web Request` built-in package to compile for iOS and Android.

> - Added new features to the `Snapshot` class [#1479](https://github.com/spatialos/gdk-for-unity/pull/1479)
>   - Adding duplicate entity ID now throws an exception

Reviewer: ```suggestion
- Adding an entity to the `Snapshot` class with a duplicate entity ID now throws an exception. [#1479](https://github.com/spatialos/gdk-for-unity/pull/1479)
```

### Added

- Added `MeansImplicitUse` attribute to `RequireAttribute` to reduce warnings in Rider IDE. [#1462](https://github.com/spatialos/gdk-for-unity/pull/1462)
- Added Event Tracing API. [#1452](https://github.com/spatialos/gdk-for-unity/pull/1452)
- Added tooltips to the SpatialOS Project Settings. [#1470](https://github.com/spatialos/gdk-for-unity/pull/1470)
- Added new features to the `Snapshot` class [#1479](https://github.com/spatialos/gdk-for-unity/pull/1479)

>   - Added method to check if snap shot contains an `EntityId`

Reviewer: ```suggestion
  - Added the `Check(EntityId entityId)` method to check if snapshot already contains an `EntityId`
```

>   - Improve search for next available `EntityId`

Reviewer: ```suggestion
  - Improved the search for next available `EntityId`, it will no longer return already used `EntityId`s.
```

>   - `Snapshot` auto-adds `Persistence` when adding entities

Reviewer: ```suggestion
  - `Snapshot` now auto-adds `Persistence` when adding entities
```

- Added `MeansImplicitUse` attribute to `RequireAttribute` to reduce warnings in Rider IDE. [#1462](https://github.com/spatialos/gdk-for-unity/pull/1462)
- Added Event Tracing API. [#1452](https://github.com/spatialos/gdk-for-unity/pull/1452)
- Added tooltips to the SpatialOS Project Settings. [#1470](https://github.com/spatialos/gdk-for-unity/pull/1470)

> - Added new features to the `Snapshot` class [#1479](https://github.com/spatialos/gdk-for-unity/pull/1479)

Reviewer: Also auto-adds `Persistence`

Reviewer: And is now `IDiposable`

  - Added method to check if `EntityId` is valid
  - Improve search for next available `EntityId`

>   - Added warning when overwriting entities

Reviewer: This now throws and as such should be in the breaking changes section
